### PR TITLE
Fix some warnings and errors on mac

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What is changed and how it works?

Issue Number: Close #xxx

What's Changed:

```commit-message

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note

```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: proxy_components/proxy_server/src/status_server/mod.rs [hunk 1]
```diff
@@ -1020,499 +1020,3 @@ where
         .body(message.into())
         .unwrap()
 }
-
-#[cfg(test)]
-mod tests {
```
Comment (JaySon-Huang) [inline proxy_components/proxy_server/src/status_server/mod.rs:1025]: why these tests are removed?

File: proxy_components/proxy_server/src/status_server/vendored_utils.rs [hunk 2]
```diff
@@ -50,6 +50,25 @@ pub fn deactivate_prof() -> ProfResult<()> {
...
+    {
+        unsafe {
+            let err = *libc::__errno_location();
+            let err_msg = libc::strerror(err);
+            let c_str = std::ffi::CStr::from_ptr(err_msg);
+            let str_slice = c_str.to_str().unwrap_or("Unknown error");
+            tikv_util::warn!(
+                "{} returns non-zero {} error_code: {} error_message: {}",
+                s,
+                r,
```
Comment (JaySon-Huang) [inline proxy_components/proxy_server/src/status_server/vendored_utils.rs:64]: where is `s` and `r` defined?

File: proxy_components/proxy_server/src/status_server/mod.rs (hunk 1 above)
Comment (CalvinNeo) [reply proxy_components/proxy_server/src/status_server/mod.rs:1025]: They don't work now. Old tests copied from TiKV

File: proxy_components/proxy_server/src/status_server/vendored_utils.rs (hunk 2 above)
Comment (CalvinNeo) [reply proxy_components/proxy_server/src/status_server/vendored_utils.rs:64]: Fixed

**Pull request merged**